e2e-tests/src/test/java/gov/nasa/jpl/aerie/e2e/utils/GatewayRequests.java

```diff
@@ -97,4 +97,36 @@ public int uploadJarFile(String jarPath) throws IOException {
       return bodyJson.getInt("id");
     }
   }
+
+
+  public void uploadExternalSourceEventTypes(JsonObject schema) throws RuntimeException {
+    final var response = request.post("/uploadExternalSourceEventTypes", RequestOptions.create()
+                                                                                       .setHeader("Authorization", "Bearer "+token)
+                                                                                 .setHeader("Content-Type", "application/json")
+                                                                                 .setData(schema.toString()));
+    // Process Response
+    try(final var reader = Json.createReader(new StringReader(response.text()))){
+      final JsonObject bodyJson = reader.readObject();
+      if(!(bodyJson.containsKey("createExternalEventTypes") && bodyJson.containsKey("createExternalSourceTypes"))){
+        System.err.println("Upload failed");
+        throw new RuntimeException(bodyJson.toString());
+      }
+    }
+  }
+
+  public void uploadExternalSource(JsonObject externalSource) throws RuntimeException {
+    final var response = request.post("/uploadExternalSource", RequestOptions.create()
+                                                                             .setHeader("Authorization", "Bearer "+token)
+                                                                                 .setHeader(
+                                                                                     "Content-Type",
+                                                                                     "application/json")
+                                                                                 .setData(externalSource.toString()));
+    try (final var reader = Json.createReader(new StringReader(response.text()))) {
+      final JsonObject bodyJson = reader.readObject();
+      if (!bodyJson.containsKey("createExternalSource")) {
+        System.err.println("Upload failed");
```

Comment:
TODO: These GatewayRequests will need to send files rather than json

Reply:
This has now been addressed!